test/e2e/dynamic_provisioning_test.go

```diff
@@ -734,6 +734,60 @@ func (t *dynamicProvisioningTestSuite) defineTests(isMultiZone bool, schedulerNa
 		}
 		test.Run(cs, ns, schedulerName)
 	})
+
+	ginkgo.It("Should test pod failover with cordoning a node", func() {
+		skipIfUsingInTreeVolumePlugin()
+		skipIfNotUsingCSIDriverV2()
+
+		t := dynamicProvisioningTestSuite{}
+
+		volume := testsuites.VolumeDetails{
+			ClaimSize: "10Gi",
+			VolumeMount: testsuites.VolumeMountDetails{
+				NameGenerate:      "test-volume-",
+				MountPathGenerate: "/mnt/test-",
+			},
+		}
+		pod := testsuites.PodDetails{
+			Cmd: convertToPowershellorCmdCommandIfNecessary("echo 'hello world' >> /mnt/test-1/data && while true; do sleep 3600; done"),
+			Volumes: t.normalizeVolumes([]testsuites.VolumeDetails{
+				{
+					ClaimSize: volume.ClaimSize,
+					MountOptions: []string{
+						"barrier=1",
+						"acl",
+					},
+					VolumeMount: volume.VolumeMount,
+				},
+			}, false),
+			IsWindows: isWindowsCluster,
+			UseCMD:    false,
+		}
+		podCheckCmd := []string{"cat", "/mnt/test-1/data"}
+		expectedString := "hello world\n"
+		if isWindowsCluster {
+			podCheckCmd = []string{"cmd", "/c", "type C:\\mnt\\test-1\\data.txt"}
+			expectedString = "hello world\r\n"
+		}
+
+		storageClassParameters := map[string]string{"skuName": "StandardSSD_LRS"}
+		if isUsingCSIDriverV2 {
+			storageClassParameters["maxShares"] = "2"
```

Comment:
We actually need to have two test cases for the v2 driver. The 1st test case leaves `maxShares` unset just like for v1; the 2nd test case sets `maxShares` to 2. The v2 driver should handle failover w/o disk sharing, and we should validate that it does.